core/renderers/measurables/inputs.js

```diff
@@ -87,36 +87,26 @@ Blockly.blockRendering.InlineInput = function(constants, input) {
       constants, input);
   this.type |= Blockly.blockRendering.Types.INLINE_INPUT;
 
-  this.setShapeDimensions(
-    !this.isDynamicShape ? this.shape.height : 0,
-    !this.isDynamicShape ? this.shape.width : 0);
-
-  this.connectionOffsetY = this.constants_.TAB_OFFSET_FROM_TOP;
-};
-Blockly.utils.object.inherits(Blockly.blockRendering.InlineInput,
-    Blockly.blockRendering.InputConnection);
-
-
-/**
- * Sets properties that depend on the connection shape dimensions.
- * @param {number} height Height of the connection shape.
- * @param {number} width Width of the connection shape.
- */
-Blockly.blockRendering.InlineInput.prototype.setShapeDimensions = function(
-    height, width) {
   if (!this.connectedBlock) {
     this.height = this.constants_.EMPTY_INLINE_INPUT_HEIGHT;
-    this.width = width +
-        this.constants_.EMPTY_INLINE_INPUT_PADDING;
+    this.width = this.constants_.EMPTY_INLINE_INPUT_PADDING;
   } else {
     // We allow the dark path to show on the parent block so that the child
     // block looks embossed.  This takes up an extra pixel in both x and y.
     this.width = this.connectedBlockWidth;
     this.height = this.connectedBlockHeight;
   }
-  this.connectionHeight = height;
-  this.connectionWidth = width;
+
+  this.connectionHeight = this.shape.height;
+  this.connectionWidth = !this.isDynamicShape ? this.shape.width :
+    this.shape.width(this.height);
+  if (!this.connectedBlock) {
```

Comment:
Merge this with the if statement on line 92.

Reply:
The connectionWidth depends on the``this.height``. And the hight is only calculated once we're through with that statement.